File: src/utils/errorHandler.js
/**
 * Centralized error handling utility
 * Logs errors and returns standardized error objects
 */

// Import winston for logging
const winston = require('winston');

// Configure logger
const logger = winston.createLogger({
  level: 'info',
  format: winston.format.combine(
    winston.format.timestamp(),
    winston.format.json()
  ),
  transports: [
    new winston.transports.File({ filename: 'error.log', level: 'error' }),
    new winston.transports.File({ filename: 'combined.log' })
  ]
});

// Add console transport in development
if (process.env.NODE_ENV !== 'production') {
  logger.add(new winston.transports.Console({
    format: winston.format.simple()
  }));
}

/**
 * Handle and log errors
 * @param {Error} error - The error object
 * @param {string} context - The context where the error occurred
 * @returns {Object} Standardized error response
 */
export const handleError = (error, context) => {
  logger.error(`Error in ${context}: ${error.message}`, {
    context,
    stack: error.stack,
    timestamp: new Date().toISOString()
  });
  
  return {
    success: false,
    error: error.message || 'An unexpected error occurred',
    context,
    timestamp: new Date().toISOString()
  };
};

/**
 * Async error wrapper for event handlers
 * @param {Function} fn - The async function to wrap
 * @returns {Function} Wrapped function with error handling
 */
export const asyncErrorHandler = (fn) => {
  return async function(...args) {
    try {
      return await fn(...args);
    } catch (error) {
      return handleError(error, fn.name || 'anonymous');
    }
  };
};

export default {
  handleError,
  asyncErrorHandler,
  logger
};

File: src/utils/error_handler.py
from loguru import logger
import sys
import traceback

# Configure logger
logger.add(
    "logs/error.log",
    rotation="500 MB",
    retention="10 days",
    level="ERROR",
    backtrace=True,
    diagnose=True
)

class ErrorHandler:
    @staticmethod
    def handle_error(error, context=""):
        """Handle and log errors with context"""
        error_type = type(error).__name__
        error_message = str(error)
        stack_trace = traceback.format_exc()
        
        error_data = {
            "type": error_type,
            "message": error_message,
            "context": context,
            "stack_trace": stack_trace
        }
        
        logger.error(f"Error in {context}: {error_message}\n{stack_trace}")
        return error_data
    
    @staticmethod
    def format_error_response(error_data):
        """Format error data for API response"""
        return {
            "success": False,
            "error": {
                "type": error_data["type"],
                "message": error_data["message"],
                "context": error_data["context"]
            }
        }